pkgs/cosmic-ext-applet-system-monitor/package.nix

```
@@ -0,0 +1,66 @@
+{
+  lib,
+  fetchFromGitHub,
+  libcosmicAppHook,
+  rustPlatform,
+  just,
+  stdenv,
+  nix-update-script,
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "cosmic-ext-applet-system-monitor";
+  version = "0.1.2";
+
+  src = fetchFromGitHub {
+    owner = "D-Brox";
+    repo = "cosmic-ext-applet-system-monitor";
+    rev = "8d133a4d5ee431b878b482f18b98e86b224992e8";
+    hash = "sha256-AwpZ2nb3YPfOAFpkoKZrDm4264UmVH96SmKLc0l81R4=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    allowBuiltinFetchGit = true;
+  };
+
+  useFetchCargoVendor = true;
+  cargoHash = "";
+
+  nativeBuildInputs = [
+    libcosmicAppHook
+    just
+  ];
+
+  dontUseJustBuild = true;
+  dontUseJustCheck = true;
+
+  justFlags = [
+    "--set"
+    "prefix"
+    (placeholder "out")
+    "--set"
+    "env-dst"
+    "${placeholder "out"}/etc/profile.d/cosmic-ext-applet-system-monitor.sh"
+    "--set"
+    "bin-src"
+    "target/${stdenv.hostPlatform.rust.cargoShortTarget}/release/cosmic-ext-applet-system-monitor"
+  ];
+
+  preCheck = ''
+    export XDG_RUNTIME_DIR="$TMP"
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    homepage = "https://github.com/D-Brox/cosmic-ext-applet-system-monitor";
+    description = "System monitor for the COSMIC Desktop Environment";
```

```suggestion
    description = "Highly configurable system resource monitor for the COSMIC Desktop Environment";
```

Let's use something closer to the upstream description in their readme